### PR TITLE
BHoM_Engine: Update MethodBase ToText to include denotation of hidden inputs

### LINKED ISSUE #3053: BHoM_Engine: Identify hidden inputs for methods
In line with https://github.com/BHoM/BHoM/pull/1506, https://github.com/BHoM/BHoM_Engine/pull/3052, and https://github.com/BHoM/BHoM_UI/pull/456, we want to highlight to users that an input is hidden.

https://github.com/BHoM/BHoM_Engine/blob/develop/BHoM_Engine/Convert/ToText.cs#L88

This is the method that needs modifying to achieve this.

Inputs should be extended to take an additional bool asking whether to include hidden inputs, defaulting to true, and additional inputs to define the start and end of a hidden input display, defaulting to `{` and `}` respectively.

### PR DESCRIPTION
**Pull request by FraserGreenroyd**
### Issues addressed by this PR
Fixes #3053

### CODE REVIEW

**Comment by FraserGreenroyd** [comment]
@BHoMBot check installer

**Comment by FraserGreenroyd** [comment]
@BHoMBot check installer

**Comment by FraserGreenroyd** [comment]
@BHoMBot check compliance

File: BHoM_Engine/Convert/ToText.cs [hunk 1]
```diff
@@ -84,8 +85,11 @@ public static string ToText(this MemberInfo member, bool includePath = false)
...
+        [Input("includeHidden", "If true, hidden inputs to the method (set using UIExposure.Hidden) will be included if the maxParams and maxChars allow.")]
+        [Input("hiddenStart", "Symbol used to separate hidden input parameters from required input parameters. Usually '{'.")]
+        [Input("hiddenEnd", "Symbol used for the end of separation of hidden input parameters from required input parameters. Usually '}'.")]
         [Output("Text representation.")]
-        public static string ToText(this MethodBase method, bool includePath = false, string paramStart = "(", string paramSeparator = ", ", string paramEnd = ")", bool removeIForInterface = true, bool includeParamNames = true, int maxParams = 5, int maxChars = 40, bool includeParamPaths = false)
+        public static string ToText(this MethodBase method, bool includePath = false, string paramStart = "(", string paramSeparator = ", ", string paramEnd = ")", bool removeIForInterface = true, bool includeParamNames = true, int maxParams = 5, int maxChars = 40, bool includeParamPaths = false, bool includeHidden = true, string hiddenStart = "{", string hiddenEnd = "}")
```
Comment (FraserGreenroyd) [inline BHoM_Engine/Convert/ToText.cs:92]:
```suggestion
        [PreviousVersion("6.2", "BH.Engine.Base.Convert.ToText(System.Reflection.MethodBase, System.Boolean, System.String, System.String, System.String, System.Boolean, System.Boolean, System.Int32, System.Int32, System.Boolean)")]
        public static string ToText(this MethodBase method, bool includePath = false, string paramStart = "(", string paramSeparator = ", ", string paramEnd = ")", bool removeIForInterface = true, bool includeParamNames = true, int maxParams = 5, int maxChars = 40, bool includeParamPaths = false, bool includeHidden = true, string hiddenStart = "{", string hiddenEnd = "}")
```

**Comment by FraserGreenroyd** [comment]
@BHoMBot check versioning

**Comment by FraserGreenroyd** [comment]
@BHoMBot check compliance
@BHoMBot check required

**Comment by FraserGreenroyd** [comment]
@BHoMBot check unit-tests

**Review by IsakNaslundBh — APPROVED** [review]
Happy with this. Code changes makes sense, and functionality tested in Grasshopper and Excel.

Happy to approve!

**Comment by FraserGreenroyd** [comment]
@BHoMBot check ready-to-merge

**Pull request merged**